src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationService.kt

```diff
@@ -0,0 +1,207 @@
+            "page" to page.first.toString(),
+          ),
+        )
+        log.error("Unable to match entire page of prisoners: $page", it)
+      }
+      .getOrElse { emptyList() }
+      .also { log.info("Page requested: $page, with ${it.size} prisoners") }
+
+  suspend fun checkMatch(prisonerId: PrisonerId): MismatchCaseNote? {
+    val offenderNo = prisonerId.offenderNo
+    return runCatching {
+      val nomisCaseNotes = caseNotesNomisApiService.getCaseNotesForPrisoner(offenderNo)
+        .caseNotes
+        .map { c ->
+          CommonCaseNoteFields(
+            c.caseNoteText,
+            c.caseNoteType.code,
+            c.caseNoteSubType.code,
+            c.occurrenceDateTime,
+            c.authorUsername,
+            c.amendments.map { a ->
+              CommonAmendmentFields(
+                text = a.text,
+                occurrenceDateTime = a.createdDateTime,
+                authorUsername = a.authorUsername,
+              )
+            },
+          )
+        }
+        .toSortedSet(fieldsComparator)
+
+      val dpsCaseNotes = caseNotesDpsApiService.getCaseNotesForPrisoner(offenderNo)
+        .filter { !it.sensitive }
+        .map { c ->
+          CommonCaseNoteFields(
+            c.text,
+            c.type,
+            c.subType,
+            c.occurrenceDateTime.format(DateTimeFormatter.ISO_DATE_TIME),
+            c.authorUsername,
+            c.amendments.map { a ->
+              CommonAmendmentFields(
+                text = a.additionalNoteText,
+                occurrenceDateTime = a.creationDateTime?.format(DateTimeFormatter.ISO_DATE_TIME),
+                authorUsername = a.authorUserName,
+              )
+            },
+          )
+        }
+        .toSortedSet(fieldsComparator)
+
+      val pairedCaseNotes = dpsCaseNotes.zip(nomisCaseNotes)
+      return if (dpsCaseNotes.size != nomisCaseNotes.size || pairedCaseNotes.any { (d, n) -> d != n }) {
+        val missingFromNomis = dpsCaseNotes - nomisCaseNotes
+        val missingFromDps = nomisCaseNotes - dpsCaseNotes
+        if (!missingFromDps.isEmpty() || !missingFromNomis.isEmpty()) {
+          log.info("missingFromNomis = $missingFromNomis")
+          log.info("missingFromDps   = $missingFromDps")
+          telemetryClient.trackEvent(
+            "casenotes-reports-reconciliation-mismatch",
+            mapOf(
+              "offenderNo" to offenderNo,
+              "missingFromDps" to (missingFromDps.joinToString()),
+              "missingFromNomis" to (missingFromNomis.joinToString()),
+            ),
+          )
+          MismatchCaseNote(offenderNo = offenderNo, missingFromDps = missingFromDps, missingFromNomis = missingFromNomis)
+        } else {
+          val diffs = pairedCaseNotes.filter { (a, b) -> a != b }
+          val dpsDiffs = diffs.map { it.first }
+          val diffsNomis = diffs.map { it.second }
+          log.info("diffs are dps = $dpsDiffs")
+          log.info("        nomis = $diffsNomis")
```

Comment:
What are the contents of a dpsDiff? Nothing sensitive?

Reply:
Ahhh 🤦